tests/test_trt_wrapper.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import tempfile
+import unittest
+
+import torch
+from parameterized import parameterized
+
+from monai.networks.nets import UNet
+from monai.utils import optional_import
+from tests.utils import skip_if_no_cuda, skip_if_quick, skip_if_windows
+
+TRTWrapper, has_trtwrapper = optional_import(
+    "monai.networks.trt_wrapper",
+    name="TRTWrapper",
+    descriptor="TRT wrapper is not available - check your installation!",
+)
+
+TEST_CASE_1 = ["fp32"]
+TEST_CASE_2 = ["fp16"]
+
+
+@skip_if_windows
+@skip_if_no_cuda
+@skip_if_quick
+class TestConvertToTRT(unittest.TestCase):
+
+    def setUp(self):
+        self.gpu_device = torch.cuda.current_device()
+
+    def tearDown(self):
+        current_device = torch.cuda.current_device()
+        if current_device != self.gpu_device:
+            torch.cuda.set_device(self.gpu_device)
+
+    @parameterized.expand([TEST_CASE_1, TEST_CASE_2])
+    @unittest.skipUnless(has_trtwrapper, "TensorRT wrapper is required for convert!")
+    def test_value(self, precision):
+        model = UNet(
+            spatial_dims=3,
+            in_channels=1,
+            out_channels=2,
+            channels=(2, 2, 4, 8, 4),
+            strides=(2, 2, 2, 2),
+            num_res_units=2,
+            norm="batch",
+        ).cuda()
+        with torch.no_grad(), tempfile.TemporaryDirectory() as _:
+            model.eval()
+            input_example = torch.randn(1, 1, 96, 96, 96).cuda()
+            output_example = model(input_example)
+            args: dict = {"tf32": True}
+            if precision == "fp16":
+                args["fp16"] = True
+                args["precision_constraints"] = "obey"
+
+            trt_wrapper = TRTWrapper("test_wrapper", model, input_names=["x"])
+            trt_wrapper.build_and_save(input_example, **args, builder_optimization_level=1)
+            trt_output = trt_wrapper(x=input_example)
```

Comment:
Currently, we have the `convert_to_trt` API in MONAI:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/utils.py#L861
Looks like it's functional similar to your new API, is it possible to combine them to avoid user confusion?

Thanks in advance.

Reply:
Well existing code relies on Torch-TensorRT encapsulation even with ONNX path so you would not be able to really use it in a while - as both bundles that use TRTWrapper  require latest TRT 10.3 RC builds so Torch-TensorRT would have to be rebuilt. Also my code uses polygraphy API to build the engine (and args), so it's not easy to merge the two together.  I will look into splitting engine build from encapsulation into TorchScript module so the engine build part can be common.

Reply:
Thanks, I am just trying to act as end-user to see how to use these MONAI APIs to convert 1 model / bundle to TensorRT, seems currently we have 2 APIs which will cause confusion in my opinion.
And we need to support it in the bundle utils scripts:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/scripts.py#L1719

Thanks.

Reply:
So, having a wrapper class to replace the original module has a number of advantages in terms of both ease to use (one can embed the conversion in the place of call where input example naturally exists) and functionality (fallback is possible for cases where only a limited set of shapes can be converted to TRT - that does not happen with fixed shape vision models, but rather common in speech and bio). That wrapper could use a torchscript-embedded TRT engine as you currently have though - main consideration here Is ability to use cutting edge TRT versions without having to rebuild Torch-TensorRT.

Reply:
OK, looking forward to your update. CC @binliunls .

Thanks.